src/service/headService.ts

```diff
@@ -0,0 +1,54 @@
+import { Head, Team_Head } from "../models";
+import dayjs from "dayjs";
+import gitzz from "gitzz";
+import { dateStr } from "../library";
+
+/**
+ * @유저_헤드_업데이트
+ */
+const updateUserHead = async (userId: number, username: string) => {
+  let MM: string = dayjs().format("MM").toString();
+
+  const totalCommit: number = await gitzz.getYearCommitCount(username);
+  const monthCommit: number = await gitzz.getSpecificMonthCommitCounts(
+    username
+  );
+
+  await Head.update(
+    { commitCount: totalCommit, [dateStr.date[MM]]: monthCommit },
+    {
+      where: {
+        year: dayjs().format("YYYY"),
+        userId: userId,
+      },
+    }
+  );
+};
```

Comment:
일케하면 뽑힌 머리 고려를 못하는듯
흠~ 머리 뽑는 방식이랑 디비 구조 쫌 변경시켜야겟다 ㅋ ㅋ ㅠ
일단... 생각을 쫌 해볼게여

Reply:
전 날 커밋수 가져오고 더하는거면 라이브러리에 하나 추가해야겠네유 오키!! 내일까지 해줄게!

머리카락 가닥수는 커밋한 날짜를 카운트하는게 맞아
그리고 `getSpecificMonthCommitCounts`는 해당 월에 커밋을 한 날짜수임니다ㅋㅋ 라이브러리 메서드이름들이 개판이라 한번 개선해야하는데.. 비슷한 이름인데 Total 붙어있는게 총 커밋수

Reply:
그럼 라이브러리 안쓰고 **디비에 저장되어있는 값 불러온거**랑 **오늘 했는지 안했는지만 체크해서 1을 더하는지 안더하는지**만 체크해서 하면 되려나..?! 공격쪽에서 괜찮을란가 모르겠네

Reply:
코드로 대충 해보면 요론 느낌..?!
```suggestion
/**
 * @유저_헤드_업데이트
 */
const updateUserHead = async (userId: number, username: string) => {
    let MM: string = dayjs().format("MM").toString();

  const totalCommit: number = await gitzz.getYearCommitCount(username);

  const commits = await Head.findOne({
    where: {
      userId: userId,
    },
    attributes: [dateStr.date[MM]],
  });

  const today = 1 ? (await gitzz.getDayCommitCount(username)) > 0 : 0;

  await Head.update(
    {
      commitCount: totalCommit,
      [dateStr.date[MM]]: commits[dateStr.date[MM]] + today,
    },
    {
      where: {
        year: dayjs().format("YYYY"),
        userId: userId,
      },
    }
  );
 };
```

Reply:
아마 12시에 Head update해주는거면 오늘여부가 아니라 어제 여부로 체크해야할듯ㅇ???! 12시 기준으로 어제!